packages/wallet/src/redux/sagas/__tests__/message-handler.test.ts

```diff
@@ -101,4 +102,106 @@ describe("message listener", () => {
       channelId: expect.any(String)
     });
   });
+
+  it.only("handles an update channel request", async () => {
+    const destinationA = Wallet.createRandom().address;
+    const signingAddressA = Wallet.createRandom().address;
+    const signingAddressB = Wallet.createRandom().address;
+    const destinationB = Wallet.createRandom().address;
+
+    // PUT  THIS INSIDE THE SHAREDDATA
+    const appDefinition = Wallet.createRandom().address;
+    const appData = "0x0";
+    const participants = [
+      {
+        participantId: "user-a",
+        signingAddress: signingAddressA,
+        destination: destinationA
+      },
+      {
+        participantId: "user-b",
+        signingAddress: signingAddressB,
+        destination: destinationB
+      }
+    ];
+    const allocations = [
+      {
+        token: "0x0",
+        allocationItems: [
+          {destination: destinationA, amount: "12"},
+          {destination: destinationB, amount: "12"}
+        ]
+      }
+    ];
+
+    const testChannelStore = {
```

Comment:
I think instead of manually constructing a channelStore, we should use `appState`,  `channelFromStates` and `setChannel` helper functions.